In [8]:
pip install openpyxl

Note: you may need to restart the kernel to use updated packages.


# 1주차 단톡방 과제
- 60171100 윤민혁

In [31]:
import pandas as pd


In [10]:
df1 = pd.read_excel('./data/Casualty.xlsx')

In [11]:
df1

Acc_Index  Vehicle_Reference  Casualty_Reference  Casualty_Class  \
0      197901A11AD14                  2                   1               1   
1      197901A1BAW34                  1                   1               3   
2      197901A1BFD77                  1                   1               1   
3      197901A1BFD77                  1                   2               2   
4      197901A1BFD77                  2                   3               1   
...              ...                ...                 ...             ...   
93040  197906P001185                  1                   1               3   
93041  197906P001187                  1                   1               3   
93042  197906P001188                  1                   1               1   
93043  197906P001199                  1                   1               1   
93044  197906P001202                  2                   1               1   

       Gender_of_Casualty  Age_Band_of_Casualty  Casualty_Severity  \
0                       1                    -1                  3   
1                       2                     6                  3   
2                       1                     5                  3   
3                       1                     4                  3   
4                       1                     6                  3   
...                   ...                   ...                ...   
93040                   1                     1                  3   
93041                   2                     7                  3   
93042                   1                     6                  3   
93043                   1                     7                  3   
93044                   2                     3                  3   

       Pedestrian_Location  Pedestrian_Movement  Car_Passenger  \
0                        0                    0              0   
1                       10                    5              0   
2                        0                    0              0   
3                        0                    0              1   
4                        0                    0              0   
...                    ...                  ...            ...   
93040                    5                    4              0   
93041                    5                    1              0   
93042                    0                    0              0   
93043                    0                    0              0   
93044                    0                    0              0   

       Bus_or_Coach_Passenger  Pedestrian_Road_Maintenance_Worker  \
0                           0                                   0   
1                           0                                  -1   
2                           0                                   0   
3                           0                                   0   
4                           0                                   0   
...                       ...                                 ...   
93040                       0                                  -1   
93041                       0                                  -1   
93042                       0                                   0   
93043                       0                                   0   
93044                       0                                   0   

       Casualty_Type  Casualty_Home_Area_Type  
0                104                       -1  
1                  0                       -1  
2                109                       -1  
3                109                       -1  
4                109                       -1  
...              ...                      ...  
93040              0                       -1  
93041              0                       -1  
93042            109                       -1  
93043            109                       -1  
93044              1                       -1  

[93045 rows x 14 colum

## 기록된 전체 교통사고 건수는?

In [12]:
len(df1)

93045

## 남성의 교통사고 건수는? 여성의 교통사고 건수는?

In [18]:
df1['Gender_of_Casualty'].value_counts()

 1    59778
 2    32961
-1      306
Name: Gender_of_Casualty, dtype: int64

- 남자는 59778, 여자는 32961

## 사고가 일어나는 버스 안에서 가장 위험한 사람은?

In [19]:
df1['Bus_or_Coach_Passenger'].value_counts()

 0    88722
-1     1837
 1     1371
 4      551
 3      354
 2      210
Name: Bus_or_Coach_Passenger, dtype: int64

- 0(버스에 탄 승객이 아님)과 -1(기록이 남아있지 않은 경우)를 제외하면 1로 승차중이던 승객이 가장 위험하다

## 전체 Casualty_Type 중 가장 많은 사고 종류는? 가장 적은 사고 종류는?

In [24]:
df1['Casualty_Type'].value_counts()

109    37273
0      23006
104    14305
1       6555
11      4488
19      2866
2       2291
108      754
113      710
90       377
103      247
110      143
105       30
Name: Casualty_Type, dtype: int64

- 가장 많은 사고 종류는 109번인 차량탑승자(개인 렌트카 포함)으로 총 37273건이다
- 가장 적은 사고 종류는 105번인 오토바이 - 콤비네이션 라이더 또는 조수석으로 30건이다

# 1주차 개인톡 과제
- 60171100 윤민혁

## 운전자들이 가장 많이 당하는 사고 종류는 ? 두번째로 많이 당하는 사고 종류는?

In [32]:
df2 = df1[df1['Casualty_Class']==1]
df2

Acc_Index  Vehicle_Reference  Casualty_Reference  Casualty_Class  \
0      197901A11AD14                  2                   1               1   
2      197901A1BFD77                  1                   1               1   
4      197901A1BFD77                  2                   3               1   
10     197901A1DAK71                  2                   2               1   
13     197901A1FBK75                  1                   1               1   
...              ...                ...                 ...             ...   
93037  197906P001182                  1                   1               1   
93039  197906P001183                  1                   2               1   
93042  197906P001188                  1                   1               1   
93043  197906P001199                  1                   1               1   
93044  197906P001202                  2                   1               1   

       Gender_of_Casualty  Age_Band_of_Casualty  Casualty_Severity  \
0                       1                    -1                  3   
2                       1                     5                  3   
4                       1                     6                  3   
10                      1                     5                  3   
13                      1                     5                  3   
...                   ...                   ...                ...   
93037                   1                     6                  3   
93039                   1                     6                  3   
93042                   1                     6                  3   
93043                   1                     7                  3   
93044                   2                     3                  3   

       Pedestrian_Location  Pedestrian_Movement  Car_Passenger  \
0                        0                    0              0   
2                        0                    0              0   
4                        0                    0              0   
10                       0                    0              0   
13                       0                    0              0   
...                    ...                  ...            ...   
93037                    0                    0              0   
93039                    0                    0              0   
93042                    0                    0              0   
93043                    0                    0              0   
93044                    0                    0              0   

       Bus_or_Coach_Passenger  Pedestrian_Road_Maintenance_Worker  \
0                           0                                   0   
2                           0                                   0   
4                           0                                   0   
10                          0                                   0   
13                          0                                   0   
...                       ...                                 ...   
93037                       0                                   0   
93039                       0                                   0   
93042                       0                                   0   
93043                       0                                   0   
93044                       0                                   0   

       Casualty_Type  Casualty_Home_Area_Type  
0                104                       -1  
2                109                       -1  
4                109                       -1  
10               109                       -1  
13               109                       -1  
...              ...                      ...  
93037            104                       -1  
93039            104                       -1  
93042            109                       -1  
93043            109                       -1  
93044              1                       -1  

[46327 rows x 14 colum

In [33]:
df_driver = df2[['Casualty_Type']]
df_driver

Casualty_Type
0                104
2                109
4                109
10               109
13               109
...              ...
93037            104
93039            104
93042            109
93043            109
93044              1

[46327 rows x 1 columns]

In [34]:
df_driver['Casualty_Type'].value_counts()

109    21469
104    12802
1       6502
2       2243
19      1858
113      569
103      222
90       220
108      210
11       164
110       44
105       24
Name: Casualty_Type, dtype: int64

- 가장 많이 당하는 사고는 109번인 개인용 렌트카 포함으로 21469건이다.
- 두번쨰로 많이 당하는 사고는 104번인 오토바이 라이더 또는 승객으로 12802건이다.